aten/src/ATen/native/cuda/Activation.cu

```diff
@@ -243,6 +247,160 @@ std::tuple<Tensor, Tensor> prelu_backward_cuda(const Tensor& grad_out_, const Te
   return std::tuple<Tensor, Tensor>{input_grad, weight_grad};
 }
 
+// -----------------------------------
+// rrelu
+// -----------------------------------
+template<typename scalar_t>
+inline scalar_t __device__ curand_uniform_type(curandStatePhilox4_32_10_t *state);
+
+template <>
+inline THHalf __device__ curand_uniform_type<THHalf>(curandStatePhilox4_32_10_t *state) {
```

Comment:
Don't use legacy THHalf type, use at::Half instead. There are implicit conversions between at::Half and float, so ScalarConvert is not necessary